### DISCUSSION TITLE
[Feature] Change  Musify language by App languages system setting screen

### DISCUSSION THREAD

**Discussion by Korb** [gokadzev/Musify#408] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues and confirmed that this feature is not already requested.

### Category of the Feature

UI Improvements

### Credits

Google

### Pain

The interface of each application in its own way hides in its settings the ability to change the language of its interface. Google has taken steps to simplify app localization and provide a unified interface for setting app language regardless of OS language.

### Suggested Solution

I suggest you consider changing the language of the application through a single control center for the languages of installed applications, available in the latest versions of Google Android. Read more about it in the article "[Change app language on your Android phone](https://support.google.com/android/answer/12395118)" of the official OS help. [Screenshot](https://drive.google.com/file/d/1rhT3cFpo8cpeLrXIPteFH-0202ks_IYy/view)

**Comment by gokadzev** [comment]
Officially, Flutter doesn't support it yet, but I've completed half of the implementation while testing. I just don't see the reason for doing this when we can change the app language from within the app itself.

**Comment by Korb** [comment]
@gokadzev, Personally, I think the best place to select the language of an application is in its properties screen (where its version, space occupied, permissions granted, power saving mode, etc. are indicated). However, Google’s option is also not bad, since it has two advantages over the “traditional” language setting in the application itself:

1. The interface of the applications is different, and the user generally has to look for a place in the settings of each application where he can change the language of the application. At the same time, there may not be any language settings in the application at all. In addition, in general, if a user wants to change the application language from English to one that he knows better, it is more convenient for him to do this through a familiar operating system interface than through the interface of an unfamiliar application displayed in a language he is not familiar with.

2. Perhaps the unified system of locales promoted by the OS developer will be more convenient for translators than alternative methods. I was prompted to this idea by the phrase

> Google has added support for inline translation with Android 12 and it may be possible for Android to translate apps on its own even if the app developers don't add other languages in the future.

from [here](https://timesofindia.indiatimes.com/gadgets-news/google-may-give-android-users-the-option-to-choose-languages-in-every-app/articleshow/88457759.cms), and the final part of the article "[Working with Android Per-App Language Preferences](https://phrase.com/blog/posts/android-per-app-language-preferences/)".

But, of course, the last word is yours.
